### PR TITLE
Bugfix for build process and removal of unused variables

### PR DESCRIPTION
**Pull request by duylethanh**
This PR contains the following:
* small bug fix regarding the phase handling during the build process
* ~~prefix directories are appended to the PATH environment variable~~
* removal of unused/unnecessary variables
* added `default_workdir` keyword in `do_step()` call in the regenerate phase to make the code consistent with the other phases

### CODE REVIEW

**Comment by avdgrinten** [comment]
This PR needs a more specific title.

File: simexpal/launch/common.py [hunk 1]
```diff
@@ -139,6 +139,7 @@ def get_paths(self):
 		paths = []
 		for build_yml in self.yml['builds']:
 			paths.append(os.path.join(build_yml['prefix'], 'bin'))
+			paths.append(build_yml['prefix'])
```
Comment (avdgrinten) [inline simexpal/launch/common.py:142]: Why this change? That seems strange to me.

Comment (duylethanh) [reply simexpal/launch/common.py:142]: There were cases where users had their executables directly in the prefix-dir.

Comment (avdgrinten) [reply simexpal/launch/common.py:142]: But isn't the fix to move them into `bin/` then? Or, we could add a `extra_paths` argument to allow the the user to configure this behavior. In any case, adding the prefix directory unconditionally does not sound like a good resolution to me.

Comment (duylethanh) [reply simexpal/launch/common.py:142]: Ok, that sounds reasonable. I will remove this line and create an issue regarding the `extra_paths` argument so that I do not forget about it and implement it in the future.

**Comment by duylethanh** [comment]
PR is ready to be merged.

**Comment by avdgrinten** [comment]
Thank you!
